injected/src/features/web-compat.js

```diff
@@ -21,9 +21,18 @@ const MSG_DEVICE_ENUMERATION = 'deviceEnumeration';
 
 function canShare(data) {
     if (typeof data !== 'object') return false;
-    if (!('url' in data) && !('title' in data) && !('text' in data)) return false; // At least one of these is required
+    // Make an in-place shallow copy of the data
```

Comment:
This just avoids having to do all of the checking on every if statement below. Including the "at least one of these" if statement.